File: LAb7/pixels.py
import sys
import numpy as np

# Taking inputs 
m, n = map(int, input("Enter value of m and n with space in between: ").split())
T = int(input("Enter value of threshold(0<=T<=255): "))

print("Readings(Enter bw 0 and 255): \n")
# Read each row line-by-line m times
img = np.array([input().split() for _ in range(m)], dtype=int)

# Converting vector entries to HIGH or LOW
bw = np.where(img >= T, 255, 0)

# Print the resulting matrix
np.savetxt(sys.stdout, bw, fmt="%d")

File: LAb7/pixels.c
#include <stdio.h>

int main() {
    int m, n, T;
    int image[100][100];
	
    printf("Enter value of m and n with space in between: ");
    scanf(" %d %d", &m, &n);
    printf("Enter value of threshold(0<=T<=255)");
    scanf("%d", &T); //taking input
    
    for(int i=0;i<m;i++){
        for(int j=0;j<n;j++){ //forming matrix of pixels
            scanf("%d", &image[i][j]);
            if(image[i][j]>= T){
                image[i][j]= 255;
            }
            else if(image[i][j]<T){
                image[i][j] =0;
            }
	    printf("\n");
        }
    }
    
    for(int i =0;i<m;i++){
        for(int j=0;j<n;j++){
            printf("%d",image[i][j]); //prints 255 or 0 
            if(j!=n-1) printf(" "); 
        }
        printf("\n");
    }

    return 0;
}
